### ISSUE TITLE
Rule suggestion: $derived should not return a function

### ISSUE THREAD

**Issue by dimfeld** [sveltejs/eslint-plugin-svelte#1332]
### Motivation

AI Coding agents sometimes get a bit confused writing Svelte 5 and create code like this:

```js
let x = $derived(() => values.filter((item) => item.active));
```

This, of course, makes `x` into a function instead of the value, and the coding agents often work around this by just changing the uses of `x` to `x()`, instead of fixing it properly to use derived.by or just a plan single-expresssion $derived.

It would be helpful to have a lint rule for this.

### Description

A $derived rune should not return a function. Either use a single expression with $derived or use $derived.by with a function.

### Examples

```svelte
<script>
// ✓ GOOD 
let x = $derived(expression);
let x = $derived.by(() => expression);
let x = $derived.by(() => {
  return complexExpression;
});
let x = $derived(calculate(value));
let x = $derived.by(calculate);

// ✗ BAD
let x =$derived(() => y);
let x =$derived(function() { return  y });
```
</script>

**Comment by dimfeld** [comment]
I had Claude Code create a proof of concept of this. Coudl use some cleanup and types but it seems to work well in my initial testing https://gist.github.com/dimfeld/04d96e8315186f42939ee534b0be05ec

**Comment by ota-meshi** [comment]
Thank you for proposing the rule.

I think a rule that reports non-reactive `$derived` declarations would be more useful than simply disallowing functions.
It's similar to `no-immutable-reactive-statements` rule , but can be thought of as a Svelte v5 adaptation.
https://sveltejs.github.io/eslint-plugin-svelte/rules/no-immutable-reactive-statements/

**Comment by dimfeld** [comment]
Yeah that makes sense to me.
